Error while building tensorflow using cmake

### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: No
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:  Linux
- **TensorFlow installed from (source or binary)**: NA
- **TensorFlow version (use command below)**: NA
- **Python version**:  NA
- **Bazel version (if compiling from source)**: NA
- **GCC/Compiler version (if compiling from source)**: GCC version 6.2.1 
- **CUDA/cuDNN version**: NA
- **GPU model and memory**: NA
- **Exact command to reproduce**: `cmake ../tensorflow/contrib/cmake -DCMAKE_INSTALL_PREFIX=/usr -DLIB_INSTALL_DIR=/usr/lib -DINCLUDE_INSTALL_DIR=/usr/include ; make`


### Describe the problem
While building tensorflow using cmake, files like `tensorflow/core/example/feature.pb_text.cc` , `tensorflow/core/framework/attr_value.pb_text.cc` are generated with incorrect header includes.
(They are missing `#include <algorithm>` ). Similar issue can be found in [this pull request](https://github.com/tensorflow/tensorflow/pull/19321)
### Source code / logs
```
[ 1011s] /home/abuild/rpmbuild/BUILD/tensorflow-1.8.0/build/tensorflow/core/example/feature.pb_text.cc: In function 'void tensorflow::internal::AppendProtoDebugString(tensorflow::strings::ProtoTextOutput*, const tensorflow::Features&)':
[ 1011s] /home/abuild/rpmbuild/BUILD/tensorflow-1.8.0/build/tensorflow/core/example/feature.pb_text.cc:464:5: error: 'stable_sort' is not a member of 'std'
[ 1011s]      std::stable_sort(keys.begin(), keys.end());
[ 1011s]      ^~~
[ 1011s] /home/abuild/rpmbuild/BUILD/tensorflow-1.8.0/build/tensorflow/core/example/feature.pb_text.cc: In function 'void tensorflow::internal::AppendProtoDebugString(tensorflow::strings::ProtoTextOutput*, const tensorflow::FeatureLists&)':
[ 1011s] /home/abuild/rpmbuild/BUILD/tensorflow-1.8.0/build/tensorflow/core/example/feature.pb_text.cc:708:5: error: 'stable_sort' is not a member of 'std'
[ 1011s]      std::stable_sort(keys.begin(), keys.end());
[ 1011s]      ^~~
[ 1011s] [ 30%] Building CXX object CMakeFiles/tf_core_framework.dir/tensorflow/core/framework/attr_value.pb_text.cc.o
[ 1012s] CMakeFiles/tf_core_framework.dir/build.make:2614: recipe for target 'CMakeFiles/tf_core_framework.dir/tensorflow/core/example/feature.pb_text.cc.o' failed
[ 1012s] make[2]: *** [CMakeFiles/tf_core_framework.dir/tensorflow/core/example/feature.pb_text.cc.o] Error 1
[ 1012s] make[2]: *** Waiting for unfinished jobs....
[ 1013s] /home/abuild/rpmbuild/BUILD/tensorflow-1.8.0/build/tensorflow/core/framework/attr_value.pb_text.cc: In function 'void tensorflow::internal::AppendProtoDebugString(tensorflow::strings::ProtoTextOutput*, const tensorflow::NameAttrList&)':
[ 1013s] /home/abuild/rpmbuild/BUILD/tensorflow-1.8.0/build/tensorflow/core/framework/attr_value.pb_text.cc:698:5: error: 'stable_sort' is not a member of 'std'
[ 1013s]      std::stable_sort(keys.begin(), keys.end());
[ 1013s]      ^~~
```

--- a/tensorflow/tools/proto_text/gen_proto_text_functions_lib.cc
+++ b/tensorflow/tools/proto_text/gen_proto_text_functions_lib.cc
@@ -803,6 +803,9 @@ void Generator::Generate(const FileDescriptor& fd) {
   // Add header to cc file.
   SetOutput(&cc_);
   Print("// GENERATED FILE - DO NOT MODIFY");
+  Print();
+  Print("#include <algorithm>");  // for `std::stable_sort()`
+  Print();
   headers = {GetProtoTextHeaderName(fd, true /* impl */)};
   AddHeadersToCurrentSection(headers);
   Print();

